src/renderer/modal/modalAutoUpdateDownloaded/view.jsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { Modal } from "modal/modal";
+import { Line } from "rc-progress";
+import Link from "component/link/index";
+
+const { ipcRenderer } = require("electron");
+
+class ModalAutoUpdateDownloaded extends React.PureComponent {
+  render() {
+    const { closeModal } = this.props;
+
+    return (
+      <Modal
+        isOpen={true}
+        type="confirm"
+        contentLabel={__("Update downloaded")}
+        confirmButtonLabel={__("Update and Restart")}
+        abortButtonLabel={__("Don't Update")}
+        onConfirmed={() => {
+          ipcRenderer.send("autoUpdateAccepted");
```

Comment:
This isn't necessarily a blocker, but is there a strong reason we have to expose `ipcRenderer` inside of a component, as well as put all of the handling of these events inside of `main/index.js`? Could this component instead dispatch an `APP_QUIT` action, and a reducer could close the app? (Or the top-level component?)

Reply:
I was going back and forth on this. Technically, neither of these IPC calls have anything to do with the store, and they do not lead to DOM updates. The "autoUpdateAccepted" IPC call essentially tells the main app that it's time to shut down (not UI state) and the "autoUpdateDeclined" IPC call just tells the main process to remember that the update was declined so that it can show the extra dialog on Windows (again, not UI state; the dialog only happens after the UI is closed).

Is it typical to move code into Redux land when it is only used for its side effects, and doesn't affect anything going on in the browser window?

Reply:
I think this is fine to leave as-is, at least for now.